### PR TITLE
Added Scaffolding docs

### PR DESCRIPTION
**Pull request by Fsalker**
Relates to https://github.com/webiny/webiny-js/issues/730

### CODE REVIEW

File: docs/get-started/scaffolding.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+sidebar_label: Scaffolding
+---
+
+# Webiny Scaffold CLI
+
+### Introduction
+Sometimes when you develop applications with Webiny you will end up spending time doing common tasks that can easily become repetitive and tedious over time. This is where `webiny scaffold` comes into play.
+
+### Usage
+You will of course need to install Webiny's CLI if you haven't done so already: `npm -g install @webiny/cli`.
```
Comment (Pavel910) [inline docs/get-started/scaffolding.md]: Not only that, you need to create a project. So I would rephrase this sentence into something like `This assumes you already have a Webiny project to work on. If not, see the getting started guide (link to guide)`.

File: docs/get-started/scaffolding.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+You will be greeted with various templates to choose from:
```
Comment (Pavel910) [inline docs/get-started/scaffolding.md]: `After running "webiny scaffold", you will be prompted to choose from various templates. By default, Webiny comes with blah blaah templates, but you can create your own, or install the existing ones from NPM.`. Something like that, just to give people more context. And maybe create another guide `How to create scaffolding templates` and link here.

Comment (Fsalker) [reply docs/get-started/scaffolding.md]: @Pavel910 here you go

**Comment by Fsalker** [comment]
The `How to create scaffolding templates` is a nice idea. I could make a subsection within `Get Started` about it & contain both this introductory part aswell as how to make templates.

I'll do this 'soon'.

**Comment by Pavel910** [comment]
I think you can put this `How to...` in this here section, with other guides, because it's an "advanced" feature, so we don't pollute the `Get started` section.

image

**Comment by Fsalker** [comment]
That's an even better idea. I was getting a bit lazy myself to click the sidebar so many times before opening the guide, especially in an early section.

File: docs/other/guides/how-to-create-scaffolding-templates.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+---
+id: creatingScaffoldingTemplates
+title: How to create Scaffolding Templates
+sidebar_label: Creating Scaffolding Templates
+---
+
+### Introduction
+As discussed in [Scaffolding](/docs/get-started/scaffolding), it is possible to save yourself a lot of time by using one of the default Scaffolding Templates.
```
Comment (Pavel910) [inline docs/other/guides/how-to-create-scaffolding-templates.md]: `it is possible to save yourself a lot of time` -> `you can save a lot of time`.
`by using one of the default Scaffolding Templates` -> `by using scaffolding templates`

File: docs/other/guides/how-to-create-scaffolding-templates.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+The great news are that you may also write your own templates! In this article we will go in depth about it.
```
Comment (Pavel910) [inline docs/other/guides/how-to-create-scaffolding-templates.md]: `The great thing is you can create your own templates`

File: docs/other/guides/how-to-create-scaffolding-templates.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+### Writing your first Scaffolding Template
+We use `node modules` for all of our templates. You should create a new package whose name follows the following pattern: `webiny-cli-scaffold-*`.
```
Comment (Pavel910) [inline docs/other/guides/how-to-create-scaffolding-templates.md]: `We use node modules for all of our templates` - took me a while to understand what you meant by this. I'd rephrase it to `Webiny CLI loads templates that are available as dependencies of your project. Under the hood, we use "yarn list" to find the available template packages.`

File: docs/other/guides/how-to-create-scaffolding-templates.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+For example, valid names could be:
+* webiny-cli-scaffold-append-unicorns
+* webiny-cli-scaffold-add-delicious-donuts-to-my-site
+
+Next, your package must export an array of [Plugins](/docs/deep-dive/plugins-crash-course) (a single Plugin is just as fine). Your Plugins must follow the object structure below:
```
Comment (Pavel910) [inline docs/other/guides/how-to-create-scaffolding-templates.md]: `a single Plugin is just as fine` -> `you can also export a single plugin`.

File: docs/other/guides/how-to-create-scaffolding-templates.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+```
+{
+    name: String,           // Plugin info
+    description: String,    // Plugin info
+    scaffold: {
+        name: String        // Used in the Terminal
+        questions({ context }): Function:[InquirerQuestion]    // Can be just an array aswell
+        generate({ input, context, oraSpinner }): async Function  
+    }
+}
```
Comment (Pavel910) [inline docs/other/guides/how-to-create-scaffolding-templates.md]: This whole kinda-TS-type section: I would prefer if it was actual TS, with valid TS types. Do we have types for these template plugins? Would be great to actually have types in the `cli` package (I know that we don't use TS in that package, but types for plugins would be very very useful). Maybe add a `cli/types.ts` file, like all other projects do in `src/types.ts` (here we don't have `src` so  add it to the root of the package).

For `InquirerQuestion` you can install `@types/inquirer` package, that will provide you with inquirer TS declarations.

File: docs/other/guides/how-to-create-scaffolding-templates.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+Additional details about the received arguments:
+* **input**: all Inquirer values will be stored here. If you asked someone for their package's name `packageName` in your `scaffold.questions`, you would access it as `input.packageName`
+* **context**: provides you the Scaffolding context filled with useful variables such as:
+    * `packagesPath` (ie: `./packages`)
+    * `apiPath`
+    * `apiYaml` (ie: `./api/serverless.yml`)
+    * `appPath` 
+    * `appYaml`
+* **oraSpinner**: the nice spinner's instance, which shows up when you run `webiny scaffold`. You are able to change its `text`, `prefixText`, `stop()` it and so on.
```
Comment (Pavel910) [inline docs/other/guides/how-to-create-scaffolding-templates.md]: Link to the library, so users can view the official `ora` docs without googling.

File: docs/other/guides/how-to-create-scaffolding-templates.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+When you have finished writing your package, the last step you have to do is installing it in your app's repository. Thus, it will be detected by Scaffold CLI and you'll be ready to go!
```
Comment (Pavel910) [inline docs/other/guides/how-to-create-scaffolding-templates.md]: `the last step you have to do is installing it in your app's repository` - `the last step is installing it in your Webiny project`. Try removing words that don't add much to the message, like `you have to do`. Since you already mentioned it's `the last step` - obviously you have to do it ;)

File: docs/get-started/scaffolding.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+After picking one of them, you might be asked a couple more questions in order to bootstrap your template. For example, a GraphQL Apollo Service template could ask for the name of your service:
+```
+? The name of your service
+```
+
+Once that's finished, your template should have been generated succesfully. As simple as that!
+
+### Default Templates
+* GraphQL Apollo Service (in PR)
```
Comment (Pavel910) [inline docs/get-started/scaffolding.md]: You can skip these `(in PR)` and `WIP` since we know you're working on it, and these docs won't be published until those packages are available on npm. So feel free to type stuff in the docs that is not yet implemented, if it's something you're already working on. This way you won't have to clean it up later.

File: docs/other/guides/how-to-create-scaffolding-templates.md (hunk 3 above)
Comment (Fsalker) [reply docs/other/guides/how-to-create-scaffolding-templates.md]: done

File: docs/other/guides/how-to-create-scaffolding-templates.md (hunk 4 above)
Comment (Fsalker) [reply docs/other/guides/how-to-create-scaffolding-templates.md]: done

File: docs/other/guides/how-to-create-scaffolding-templates.md (hunk 5 above)
Comment (Fsalker) [reply docs/other/guides/how-to-create-scaffolding-templates.md]: yeah, they aren't ruby gems, python packages or whatever else...

but since we only use node i suppouse it's redundant to mention taht

Comment (Fsalker) [reply docs/other/guides/how-to-create-scaffolding-templates.md]: done

File: docs/other/guides/how-to-create-scaffolding-templates.md (hunk 6 above)
Comment (Fsalker) [reply docs/other/guides/how-to-create-scaffolding-templates.md]: done (this falls in the same category as the 1st comment i believe)

File: docs/other/guides/how-to-create-scaffolding-templates.md (hunk 8 above)
Comment (Fsalker) [reply docs/other/guides/how-to-create-scaffolding-templates.md]: done

File: docs/other/guides/how-to-create-scaffolding-templates.md (hunk 9 above)
Comment (Fsalker) [reply docs/other/guides/how-to-create-scaffolding-templates.md]: i have yet to become an expert writer.

by writing these docs i'm becoming a pro in no time

Comment (Fsalker) [reply docs/other/guides/how-to-create-scaffolding-templates.md]: done

File: docs/get-started/scaffolding.md (hunk 10 above)
Comment (Fsalker) [reply docs/get-started/scaffolding.md]: ah, great to know

done

File: docs/other/guides/how-to-create-scaffolding-templates.md (hunk 7 above)
Comment (Fsalker) [reply docs/other/guides/how-to-create-scaffolding-templates.md]: done

File: docs/get-started/scaffolding.md (hunk 1 above)
Comment (Fsalker) [reply docs/get-started/scaffolding.md]: (done a while ago; checking over again right now)

**Comment by Pavel910** [comment]
@Fsalker is this PR still relevant? I can see `serverless.yml` in the preview deploy, not sure what state this is in 🤔
